### PR TITLE
9899 add location notes

### PR DESCRIPTION
**Pull request by roseeichelmann**
## Associated issues

Closes cityofaustin/atd-data-tech/issues/9899
Closes cityofaustin/atd-data-tech/issues/9900
Closes cityofaustin/atd-data-tech/issues/9901

This PR creates a Notes section in the Location details that has the same functionality as the Crash details Notes feature. Users with edit & admin permissions should be able to read, add, edit, and delete notes. Users with only read permissions should only be able to read existing notes.
 
## Testing
**URL to test:** 

This can be tested locally or as a Netlify preview (https://deploy-preview-1090--atd-vze-staging.netlify.app/#/dashboard)

**Steps to test:**

1. Go to Locations tab.
2. Click on Location ID 644AF55CFC (https://deploy-preview-1090--atd-vze-staging.netlify.app/#/locations/644AF55CFC)
3. New Notes section should be underneath aerial map and details components.
4. If you have edit permissions, add a new note using add button. New note should appear under the add note row.
5. The edit button (pencil) should only appear for notes you created (updated by column).
6. Click edit button on a note you created then cancel (X button). Note should be unchanged.
7. Click edit button, change nothing, then save (check button). Note should be unchanged.
8. Click edit button, change the note, then save. Note should be updated.
9. Delete your note.
10. Go to Location ID 8FCA19BF67 or 0A4FCA4EA6 to see other Notes sections with dummy data, repeat steps if you wish!

---
#### Ship list
- [x] Code reviewed
- [x] Product manager approved

Note: For deployment, the following steps will need to be done in the Production Hasura console:
- [ ] Rename "notes" table "crash_notes"
- [ ] Add a location notes table called "location_notes"
- [ ] Add date, user_email, created_at, updated_at, text, id, location id columns to "location_notes" with the same parameters as staging console.

### CODE REVIEW

**Review by patrickm02L — APPROVED** [review]
Looks good! 🤘

Screen Shot 2022-08-04 at 7 10 33 AM

**Review by patrickm02L — APPROVED** [review]
It's a go! 🤘

https://user-images.githubusercontent.com/5566727/182894788-83be51a3-c47e-44dc-8371-516d47934d66.mp4

File: atd-vze/src/views/Locations/LocationNotes.js [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+  // function to handle cancel button click
+  const handleCancelClick = () => {
+    setEditedNote("");
+		setEditRow("");
+  };
+
+  // function to handle delete note button click
+  const handleDeleteClick = (row) => {
+    const id = row.id
```
Comment (mateoclarke) [inline atd-vze/src/views/Locations/LocationNotes.js:85]: should have a semicolon at the end of this line. Prettier would do this for you.

**Review by mateoclarke — APPROVED** [review]
Looks good! Would love to see those indentations and extra console logs addressed before we merge but otherwise all good.

I also think there's a discussion we can have in our 1:1 about how we make some of the Notes code reusable between the Locations and Crashes. That way if we need to update behavior, we don't have to do it in both places.

Last thing, can you make a list of steps we'll need to do in the production Hasura console for the deployment? Any schema updates (ex: naming `notes` to `crash_notes`) we'll need to replicate from the staging env to prod.

File: atd-vze/src/views/Crashes/Notes.js [hunk 2]
```diff
@@ -67,16 +67,18 @@ const Notes = ({ crashId }) => {
         id: id
       }
     }).then(response => {
-      setEditedNote("");
-      refetch();
+			refetch().then(response => {
+        setEditedNote("");
+        setEditRow("");
```
[the remark this answers is not in the document]
Comment (roseeichelmann) [reply atd-vze/src/views/Crashes/Notes.js:72]: actually - i remember why i added the extra ```.then()```, it actually does serve some functionality, i will show you in our 1:1.

**Comment by mateoclarke** [comment]
Thanks for adding the deploy notes about db changes for the prod Hasura console. That's perfect!

**Pull request merged**